core/field.js

```diff
@@ -782,6 +770,27 @@ Blockly.Field.prototype.setValue = function(newValue) {
   doLogging && console.log(this.value_);
 };
 
+/**
+ * Process the result of validation.
+ * @param {*} newValue New value.
+ * @param {*} validatedValue Validated value.
+ * @return {*} New value, or an Error object.
+ * @private
+ */
+Blockly.Field.prototype.validate_ = function(newValue, validatedValue) {
```

Comment:
Looks like a good change!

I might rename this to something like "processValidation_" though since I believe validate_ used to be used inside the text input field, and may conflict.

Reply:
Good call.  Done.